packages/core/plugin/src/withSentryAndroidGradlePlugin.ts

```diff
@@ -0,0 +1,96 @@
+  autoUploadProguardMapping?: boolean;
+  uploadNativeSymbols?: boolean;
+  includeNativeSources?: boolean;
+  includeSourceContext?: boolean;
+  autoInstallationEnabled?: boolean;
+}
+
+/**
+ * Adds the Sentry Android Gradle Plugin to the project.
+ * https://docs.sentry.io/platforms/react-native/manual-setup/manual-setup/#enable-sentry-agp
+ */
+export function withSentryAndroidGradlePlugin(config: any, options: SentryAndroidGradlePluginOptions = {}): any {
+  const version = options.sentryAndroidGradlePluginVersion;
+  const includeProguardMapping = options.includeProguardMapping ?? true;
+  const dexguardEnabled = options.dexguardEnabled ?? false;
+  const autoUploadProguardMapping = options.autoUploadProguardMapping ?? true;
+  const uploadNativeSymbols = options.uploadNativeSymbols ?? true;
+  const autoUploadNativeSymbols = options.autoUploadNativeSymbols ?? true;
+  const includeNativeSources = options.includeNativeSources ?? true;
+  const includeSourceContext = options.includeSourceContext ?? false;
+  const autoInstallationEnabled = options.autoInstallationEnabled ?? false;
+
+  // Modify android/build.gradle
+  const withSentryProjectBuildGradle = (config: any): any => {
+    return withProjectBuildGradle(config, (projectBuildGradle: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults || !projectBuildGradle.modResults.contents) {
+        throw new Error('android/build.gradle content is missing or undefined.');
+      }
+
+      const dependency = `classpath("io.sentry:sentry-android-gradle-plugin:${version}")`;
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults.contents.includes(dependency)) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        projectBuildGradle.modResults.contents = projectBuildGradle.modResults.contents.replace(
+          /dependencies\s*{/,
+          `dependencies {\n        ${dependency}`,
+        );
+      }
+
+      return projectBuildGradle;
+    });
+  };
+
+  // Modify android/app/build.gradle
+  const withSentryAppBuildGradle = (config: any): any => {
+    return withAppBuildGradle(config, (config: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (config.modResults.language === 'groovy') {
+        const sentryPlugin = `apply plugin: "io.sentry.android.gradle"`;
+        const sentryConfig = `
+  sentry {
+      autoUploadProguardMapping = ${autoUploadProguardMapping}
+      includeProguardMapping = ${includeProguardMapping}
+      dexguardEnabled = ${dexguardEnabled}
+      uploadNativeSymbols = ${uploadNativeSymbols}
+      autoUploadNativeSymbols = ${autoUploadNativeSymbols}
+      includeNativeSources = ${includeNativeSources}
+      includeSourceContext = ${includeSourceContext}
+      autoInstallation {
+          enabled = ${autoInstallationEnabled}
+      }
```

Comment:
Let's also bake in tracing disabled to avoid any potential issues, as this is not used in RN at the moment.

```
    // Enable or disable the tracing instrumentation.
    // Does auto instrumentation for specified features through bytecode manipulation.
    // Default is enabled.
    tracingInstrumentation {
      enabled = true
```

Reply:
Disabled the tracingInstrumentation with 6dbf9b54badfe26ba3007d6e6b3e893aa87c8ea8